LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -260,20 +286,30 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                         }
                     }
 
-                    if (!isConfirmedResubmit)
+                    var sendUpstream = concentratorDeduplicationResult is ConcentratorDeduplicationResult.NotDuplicate || loRaDevice.Deduplication is not DeduplicationMode.Drop;
```

Comment:
can't we use the fcntup validation before to determine, if we want to send upstream?

Reply:
You mean add this test up when we check also the fcntup? We could do that but that would change the existing logic. Currently we are setting the `requiresConfirmation`, getting cloud to a device message (if not mistaken) and send class C messages (?), even if we skip sending to IoTHub.

Reply:
what I meant is we have this to enter the upstream handling:

`var canSendUpstream = isFrameCounterFromNewlyStartedDevice
                                  || payloadFcntAdjusted > loRaDevice.FCntUp
                                  || (payloadFcntAdjusted == loRaDevice.FCntUp && concentratorDeduplicationResult is ConcentratorDeduplicationResult.SoftDuplicateDueToDeduplicationStrategy);`

wouldn't this be the indication already that we want to send upstream?

Reply:
We can resolve this. This is needed to distinguish the case where we have a device reset or a resubmit and we still want to check the decoder result for downstream but not send upstream. The logic of the decoder validation should be extracted.